src/sysfiles.h

```diff
@@ -78,6 +78,35 @@ extern Int SyLoadModule( const Char * name, InitInfoFunc * func );
 
 /****************************************************************************
 **
+*F * * * * * * * * * * finding location of executable * * * * * * * * * * * *
+*/
+
+/* On some OSes PATH_MAX is unreliable, so we make our buffer longer.
+** No buffer size is perfect, but we need to use a hard-code size.
```

Comment:
tl;dr: I will make a PR which should allow you to use `GAP_PATH_MAX` and get rid of this comment, `LONG_PATH_SIZE_1` and `LONG_PATH_SIZE`. 

"No buffer size is perfect" is both an understatement and not quite accurate. Also: why do we "need" to use a hard-coded size? Isn't it rather that we do this for convenience? We could instead allocate dynamic buffers, and
check for `errno == ERANGE` and if that is encountered, allocate larger buffers and try again. Now, I am *not* saying you should do that. Just that the comment seems "wrong".

Also, if I read this w/o knowing the background of all this, I'd serious wonder why there is both `GAP_PATH_MAX` and `LONG_PATH_SIZE`, and which to use. An easy out to that is to move `LONG_PATH_SIZE` into the C file for now, and/or add a `FIXME: unify this with GAP_PATH_MAX`.

But I think it's actually better to "fix" `GAP_PATH_MAX` to equal `max(PATH_MAX, 4096)` (and change the few places where such a large value is problematic to use another constant, at least for now). 

I'll make a pull request for that, after which I think you can get rid of `LONG_PATH_SIZE` and just use `GAP_PATH_MAX`

Reply:
One thing to point out here -- some of the functions we are using (in particular `realpath`), don't take a buffer size, and just overflow if the buffer isn't long enough. Therefore I have to err on the size of using stupidly large buffers everywhere.

The standard method of fixing this is to pass in `NULL` and then `realpath` will allocate a buffer with malloc, but obviously that's no good for us.

Reply:
That "standard method" was not in the initial `realpath` specification and only shoehorned on later, so it's not clear all systems support it -- and trying to use this `NULL` trick on a system which doesn't support may well lead to a crash.

Anyway, if `PATH_MAX` is #defined, then `GAP_PATH_MAX >= PATH_MAX` holds with the changes in PR #1894, so that bit is fine. However, `PATH_MAX` may also *not* be defined, in accordance with POSIXm which says a system which cannot define an upper bound on path sizes should not define that. In that case, the only safe way to call `realpath` is to use that `NULL` argument.

Ahh, POSIX *sigh*